server/domain/devices/user.go

```diff
@@ -0,0 +1,48 @@
+package devices
+
+import (
+	"crypto/sha1"
+	"encoding/base64"
+	"errors"
+	"fmt"
+
+	"github.com/jinzhu/gorm"
+)
+
+type User struct {
+	gorm.Model
+	Email    string `json:"email"`
+	Password string `json:"password"`
+}
+
+func (User) TableName() string {
+	return "users"
+}
+
+func generateHash(pwd []byte) string {
+	hasher := sha1.New()
+	hasher.Write(pwd)
+	return base64.URLEncoding.EncodeToString(hasher.Sum(nil))
+}
+
+func (t *User) Get(email string, password string) (*User, error) {
+	user := new(User)
+	var hashedPWD string = generateHash([]byte(password))
```

Comment:
You should not do that here - DB methods should only do the DB calls, not the business logic